File: sandbox-platform/internal/router/rules.go
package router

import "github.com/sandbox/platform/pkg/types"

// defaultRules returns the built-in tool → tier mapping.
func defaultRules() map[string]types.Tier {
	return map[string]types.Tier{
		// WASM tier — fast, stateless tools
		"html_parse":       types.TierWASM,
		"json_parse":       types.TierWASM,
		"markdown_convert": types.TierWASM,
		"docx_generate":    types.TierWASM,
		"echo":             types.TierWASM,
		"hello":            types.TierWASM,

		// MicroVM tier — I/O, network, subprocess
		"python_run": types.TierMicroVM,
		"bash_run":   types.TierMicroVM,
		"git_clone":  types.TierMicroVM,
		"file_ops":   types.TierMicroVM,

		// GUI tier — requires a display
		"browser_open":      types.TierGUI,
		"web_scrape":        types.TierGUI,
		"excel_edit":        types.TierGUI,
		"office_automation": types.TierGUI,
	}
}

File: src/load-balancer/router/rules.py
"""Default tool-to-tier routing rules.

Mirrors internal/router/rules.go.
"""

from load_balancer.types import Tier


def default_rules() -> dict[str, Tier]:
    """Return the built-in tool → tier mapping."""
    return {
        # WASM tier — fast, stateless tools
        "html_parse": Tier.WASM,
        "json_parse": Tier.WASM,
        "markdown_convert": Tier.WASM,
        "docx_generate": Tier.WASM,
        "echo": Tier.WASM,
        "hello": Tier.WASM,
        # MicroVM tier — I/O, network, subprocess
        "python_run": Tier.MICROVM,
        "bash_run": Tier.MICROVM,
        "git_clone": Tier.MICROVM,
        "file_ops": Tier.MICROVM,
        # GUI tier — requires a display
        "browser_open": Tier.GUI,
        "web_scrape": Tier.GUI,
        "excel_edit": Tier.GUI,
        "office_automation": Tier.GUI,
    }
